src/components/characters/Characters.js

```diff
@@ -0,0 +1,51 @@
+import React, { PureComponent } from 'react';
+import Character from './Character';
+import  { getCharacters } from '../../services/rickAndMortyApi';
+import PropTypes from 'prop-types';
+import Paging from '../paging/Paging';
+
+export default class Characters extends PureComponent {
+  state = {
+    characters: [],
+    currentPage: 1,
+    totalPages: null
+  } 
+  static propTypes = {
+    updateTotalPages: PropTypes.func.isRequired,
+    increaseCount: PropTypes.func.isRequired,
+    decreaseCount: PropTypes.func.isRequired
+  }
+  componentDidMount() {
+    getCharacters(this.state.currentPage)
+      .then(res => this.setState({ characters: res.results }, () => {
+        this.props.updateTotalPages(res.totalPages);
+      }));
+  }
+  componentDidUpdate() {
```

Comment:
You need to have a guard in here to prevent yourself from getting rate-limited by the R&M API:
Something to the extent of this: 
```suggestion
  componentDidUpdate(prevProps, prevState) {
  if(prevState.currentPage !== this.state.currentPage) {
    getCharacters(this.state.currentPage)
      .then(res => this.setState({ characters: res.results }, () => {
        this.props.updateTotalPages(res.totalPages);
      }));
     }
```

And you'll need the same thing for your Locations component as well.